File: k9mail/src/test/java/com/fsck/k9/notification/AddNotificationResultTest.java
package com.fsck.k9.notification;


import org.junit.Before;
import org.junit.Test;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertFalse;
import static org.junit.Assert.assertTrue;


public class AddNotificationResultTest {
    private static final int NOTIFICATION_ID = 23;


    private NotificationHolder notificationHolder;


    @Before
    public void setUp() throws Exception {
        notificationHolder = new NotificationHolder(NOTIFICATION_ID, null);
    }

    @Test
    public void newNotification_shouldCancelNotification_shouldReturnFalse() throws Exception {
        AddNotificationResult result = AddNotificationResult.newNotification(notificationHolder);

        assertFalse(result.shouldCancelNotification());
    }

    @Test(expected = IllegalStateException.class)
    public void newNotification_getNotificationId_shouldReturnNotificationId() throws Exception {
        AddNotificationResult result = AddNotificationResult.newNotification(notificationHolder);

        result.getNotificationId();
    }

    @Test
    public void replaceNotification_shouldCancelNotification_shouldReturnTrue() throws Exception {
        AddNotificationResult result = AddNotificationResult.replaceNotification(notificationHolder);

        assertTrue(result.shouldCancelNotification());
    }

    @Test
    public void replaceNotification_getNotificationId_shouldReturnNotificationId() throws Exception {
        AddNotificationResult result = AddNotificationResult.replaceNotification(notificationHolder);

        assertEquals(NOTIFICATION_ID, result.getNotificationId());
    }

    @Test
    public void getNotificationHolder_shouldReturnNotificationHolder() throws Exception {
        AddNotificationResult result = AddNotificationResult.replaceNotification(notificationHolder);

        assertEquals(notificationHolder, result.getNotificationHolder());
    }
}

File: app/core/src/test/java/com/fsck/k9/notification/AddNotificationResultTest.kt
package com.fsck.k9.notification

import com.fsck.k9.controller.MessageReference
import com.google.common.truth.Truth.assertThat
import org.junit.Test

private const val NOTIFICATION_ID = 23

class AddNotificationResultTest {
    private val notificationHolder = NotificationHolder(
        notificationId = NOTIFICATION_ID,
        content = NotificationContent(
            messageReference = MessageReference("irrelevant", 1, "irrelevant"),
            sender = "irrelevant",
            subject = "irrelevant",
            preview = "irrelevant",
            summary = "irrelevant"
        )
    )

    @Test
    fun newNotification_shouldCancelNotification_shouldReturnFalse() {
        val result = AddNotificationResult.newNotification(notificationHolder)

        assertThat(result.shouldCancelNotification).isFalse()
    }

    @Test(expected = IllegalStateException::class)
    fun newNotification_getNotificationId_shouldReturnNotificationId() {
        val result = AddNotificationResult.newNotification(notificationHolder)

        result.notificationId
    }

    @Test
    fun replaceNotification_shouldCancelNotification_shouldReturnTrue() {
        val result = AddNotificationResult.replaceNotification(notificationHolder)

        assertThat(result.shouldCancelNotification).isTrue()
    }

    @Test
    fun replaceNotification_getNotificationId_shouldReturnNotificationId() {
        val result = AddNotificationResult.replaceNotification(notificationHolder)

        assertThat(result.notificationId).isEqualTo(NOTIFICATION_ID)
    }

    @Test
    fun getNotificationHolder_shouldReturnNotificationHolder() {
        val result = AddNotificationResult.replaceNotification(notificationHolder)

        assertThat(result.notificationHolder).isEqualTo(notificationHolder)
    }
}
